src/test/txdb_tests.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019-2020 Chaintope Inc.
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <txdb.h>
+#include <test/test_tapyrus.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(txdb_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(txdb_aggpubkey_serialize_tests)
+{
+    XFieldAggpubkey xFieldAggpubkey;
+    std::vector<XFieldAggpubkey> xFieldAggpubkeyList;
+    CDataStream stream(ParseHex("0121025700236c2890233592fcef262f4520d22af9160e3d9705855140eb2aa06c35d30a0000000000000000000000000000000000000000000000000000000000000000000000"), SER_NETWORK, PROTOCOL_VERSION);
+    stream >> xFieldAggpubkeyList;
+
+    // 1 item
+    xFieldAggpubkey = xFieldAggpubkeyList[0];
+    BOOST_CHECK_EQUAL(HexStr(xFieldAggpubkey.aggpubkey.begin(), xFieldAggpubkey.aggpubkey.begin()+xFieldAggpubkey.aggpubkey.size()), "025700236c2890233592fcef262f4520d22af9160e3d9705855140eb2aa06c35d3");
+    BOOST_CHECK_EQUAL(xFieldAggpubkey.height, 10);
+    BOOST_CHECK_EQUAL(xFieldAggpubkey.blockHash, uint256S("0000000000000000000000000000000000000000000000000000000000000000000000"));
+
+    // 2 items
+    CDataStream stream1(ParseHex("0221025700236c2890233592fcef262f4520d22af9160e3d9705855140eb2aa06c35d30a00000000000000000000000000000000000000000000000000000000000000000000002103b44f1cfcf46aba8bc98e2fd39f137cc43d98ab7792e4848b09c06198b042ca8b00000000d39be6bcedc39831203a6ab531040f9ab30d1e66c1cbcd657c38b590d91e0da7"), SER_NETWORK, PROTOCOL_VERSION);
+    stream1 >> xFieldAggpubkeyList;
+
+    xFieldAggpubkey = xFieldAggpubkeyList[1];
+    BOOST_CHECK_EQUAL(HexStr(xFieldAggpubkey.aggpubkey.begin(), xFieldAggpubkey.aggpubkey.begin()+xFieldAggpubkey.aggpubkey.size()), "03b44f1cfcf46aba8bc98e2fd39f137cc43d98ab7792e4848b09c06198b042ca8b");
+    BOOST_CHECK_EQUAL(xFieldAggpubkey.height, 0);
+    BOOST_CHECK_EQUAL(xFieldAggpubkey.blockHash, uint256S("a70d1ed990b5387c65cdcbc1661e0db39a0f0431b56a3a203198c3edbce69bd3"));
+
+    // 5 items - error
+    CDataStream stream2(ParseHex("0521025700236c2890233592fcef262f4520d22af9160e3d9705855140eb2aa06c35d30a00000000000000000000000000000000000000000000000000000000000000000000002103b44f1cfcf46aba8bc98e2fd39f137cc43d98ab7792e4848b09c06198b042ca8b00000000d39be6bcedc39831203a6ab531040f9ab30d1e66c1cbcd657c38b590d91e0da7"), SER_NETWORK, PROTOCOL_VERSION);
+
```

Comment:
What happens if the number exceeds `255(0xff)`?

Reply:
It uses VarInt to serialise and deserialise these length fields. The resulting integer would be anywhere from one byte to 5 bytes in the stream. I added a serialise test with a list of size 300 to verify this.